### PR TITLE
[cmake][addons] change binary addon includes cmake install way

### PR DESCRIPTION
**Pull request by AlwinEsch**
## Description

This change the way how the headers of binary addons becomes
installed from Kodi. Instead to define every file becomes the
folder with headers installed. All includes who are outside
from the dev-kit place are still installed by the old way.

Reasons of this changes are:
- Allow to copy header who defined in different include folders,
e.g. the coming:
```cpp
#include <kodi/addon-instance/PVRClient.h>
#include <kodi/gui/DialogOK.h>
```
- Make it more easy during add of new header in dev-kit, before was always
a update on cmake needed.

The only thing to be noticed is that the dev-kit include only headers.
This was also the main reason that this folder was made and did not give
it as hard.

## Motivation and Context
Next part from kodi-agile

## Types of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [ ] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by AlwinEsch** [comment]
Is that OK and have not forgotten anything?

File: cmake/cpack/deb/packages/kodi-addon-dev.txt.in [hunk 1]
```diff
@@ -16,7 +16,7 @@ PACKAGE_SECTION libdevel
 PACKAGE_PRIORITY optional
 PACKAGE_SHLIBDEPS
 PACKAGE_DEPENDS
-PACKAGE_RECOMMENDS @APP_NAME_LC@-audio-dev, @APP_NAME_LC@-inputstream-dev, @APP_NAME_LC@-pvr-dev, @APP_NAME_LC@-screensaver-dev, @APP_NAME_LC@-visualization-dev
+PACKAGE_RECOMMENDS
 PACKAGE_SUGGESTS
 PACKAGE_BREAKS
 PACKAGE_REPLACES
```
Comment (hudokkow) [inline cmake/cpack/deb/packages/kodi-addon-dev.txt.in]: I think `PACKAGE_REPLACES` should be populated with `@APP_NAME_LC@-audio-dev, @APP_NAME_LC@-inputstream-dev, @APP_NAME_LC@-pvr-dev, @APP_NAME_LC@-screensaver-dev, @APP_NAME_LC@-visualization-dev`

@wsnipex to confirm ^^

File: xbmc/addons/addon-bindings.mk [hunk 2]
```diff
@@ -1,45 +1,11 @@
+# List here contains all addon related header who are not defined in
+# ./addons/kodi-addon-dev-kit/include/kodi
```
Comment (hudokkow) [inline xbmc/addons/addon-bindings.mk]: List contains only add-on related headers not inside

**Review by hudokkow — COMMENTED** [review]
Looks good to me, apart from the minors pointed.
Give me a few minutes and I'll test the PR.

File: cmake/cpack/deb/packages/kodi-addon-dev.txt.in (hunk 1 above)
Comment (wsnipex) [inline cmake/cpack/deb/packages/kodi-addon-dev.txt.in]: correct. If you replace all the different packages with only kodi-addon-dev, this needs to replace the old packages

**Comment by wsnipex** [comment]
this will require changes for PPA packaging.

**Review by hudokkow — APPROVED** [review]
Tested and working 100%. Just needs to minors corrected.
:+1:

**Comment by AlwinEsch** [comment]
is updated

jenkins build this please

**Comment by AlwinEsch** [comment]
jenkins build this please

**Comment by wsnipex** [comment]
who is going to fix debian packaging in every addon?

**Comment by hudokkow** [comment]
Me? 😆

**Comment by AlwinEsch** [comment]
I can also do it :-D Am anyway in addon merge mood

**Comment by notspiff** [comment]
globbing for headers. hrmf.

**Comment by AlwinEsch** [comment]
Hi @notspiff what exactly you mean?
1. Bad to use folder instead of addon-bindings.mk
2. Better to use `file(GLOB_RECURSE` instead `file(copy`

**Comment by notspiff** [comment]
1. It's a generic principle I follow, I never glob for sources, it makes things unpredictable. No biggie, just a knee jerk. It's probably more convenient than dangerous in the context.

**Comment by AlwinEsch** [comment]
I give you partly right. But to handle includes in different destination folders has produced some problems. Also problematic becomes then that in `cmake/scripts/linux/Install.cmake` it is defined as files.

This Install.cmake is for me really a bad way and need to be improved in future. That's why I also like to have only one debian dev package there who can use addon-bindings.mk.

I can try to create a macro to read a modified addon-bindings.mk and handle them also in Install.cmake.

**Pull request merged**